### PR TITLE
fix: seed-by-insert

### LINKED ISSUE #279: Seeds don't work without new permissions since 1.4.2
Change #161 updated the way seeds work to upload to S3 and apply a schema instead of a create/insert, to support larger seed files. Unfortunately, that is a breaking change for me - I have several small seed files and no access to upload to S3.

- Is there a way of toggling between the old and new behaviour? If not I can try and PR something in to do that
- Could you do something to detect changes that increase the scope of necessary IAM permissions?  #262 is another change that required new IAM permissions that I don't have in my regulated environment

### PR DESCRIPTION
**Pull request by brabster**
### Description

resolve #279 
builds on #292 (so that I can test both in my own projects and see if they resolve my issues, can't test one without the other as they're both breaks for seeds :sob:)

Changes are actually in dbt/include/athena/macros/materializations/seeds/helpers.sql (the rest are approved but not merged from #292)
Test is tests/functional/adapter/test_seed_by_insert.py

Introduce a way to configure seeds to insert as SQL instead of upload to S3. Seems to be some duplication but not sure if/how best to tackle that

## Models used to test - Optional
New test case added

## Checklist
- [/] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [/] You kept your Pull Request small and focused on a single feature or bug fix.
- [/] You added unit testing when necessary
- [/] You added functional testing when necessary

### CODE REVIEW

File: dbt/include/athena/macros/materializations/seeds/helpers.sql [hunk 1]
```diff
@@ -31,7 +31,54 @@
     ) as {{ col }}
 {% endmacro %}
 
-{% macro athena__create_csv_table(model, agate_table) %}
+{% macro create_csv_table_insert(model, agate_table) %}
+  {%- set identifier = model['alias'] -%}
+
+  {%- set lf_tags = config.get('lf_tags', default=none) -%}
+  {%- set lf_tags_columns = config.get('lf_tags_columns', default=none) -%}
```
Comment (svdimchenko) [inline dbt/include/athena/macros/materializations/seeds/helpers.sql:38]:
```suggestion
{%- set lf_tags_config = config.get('lf_tags_config') -%}
{%- set lf_grants = config.get('lf_grants') -%}
```
pls pay attention about updates in lakeformation integration

**Comment by brabster** [comment]
Marked draft until #292 merged

File: dbt/include/athena/macros/materializations/seeds/helpers.sql [hunk 2]
```diff
@@ -31,7 +31,54 @@
...
+{% macro create_csv_table_insert(model, agate_table) %}
+  {%- set identifier = model['alias'] -%}
+
+{%- set lf_tags_config = config.get('lf_tags_config') -%}
+{%- set lf_grants = config.get('lf_grants') -%}
+  {%- set column_override = config.get('column_types', {}) -%}
+  {%- set quote_seed_column = config.get('quote_columns', None) -%}
+  {%- set s3_data_dir = config.get('s3_data_dir', default=target.s3_data_dir) -%}
+  {%- set s3_data_naming = config.get('s3_data_naming', target.s3_data_naming) -%}
+  {%- set external_location = config.get('external_location', default=none) -%}
```
Comment (svdimchenko) [inline dbt/include/athena/macros/materializations/seeds/helpers.sql:43]:
```suggestion
  {%- set lf_tags_config = config.get('lf_tags_config') -%}
  {%- set lf_grants = config.get('lf_grants') -%}
  {%- set column_override = config.get('column_types', {}) -%}
  {%- set quote_seed_column = config.get('quote_columns') -%}
  {%- set s3_data_dir = config.get('s3_data_dir', target.s3_data_dir) -%}
  {%- set s3_data_naming = config.get('s3_data_naming', target.s3_data_naming) -%}
  {%- set external_location = config.get('external_location') -%}
```
just to keep consistency

**Comment by svdimchenko** [comment]
@brabster the code looks good to me. Moreover I need this feature as well because there may be cases when parallel run of seed command and run command from different places may affect one another and break data processing pipeline. And your PR fixes this. So that thanks a lot for implementing such option!

I've left just minor recommendation to keep configs section consistent.

**Comment by brabster** [comment]
Added some documentation, ready to go for me. Thanks @svdimchenko

File: dbt/include/athena/macros/materializations/seeds/helpers.sql [hunk 3]
```diff
@@ -31,7 +31,54 @@
...
+    )
+    location '{{ location }}'
+
+  {% endset %}
+
+  {% call statement('_') -%}
+    {{ sql_table }}
+  {%- endcall %}
+
+
```
Comment (svdimchenko) [inline dbt/include/athena/macros/materializations/seeds/helpers.sql:76]:
```suggestion

  {% if lf_tags_config is not none %}
    {{ adapter.add_lf_tags(relation, lf_tags_config) }}
  {% endif %}

  {% if lf_grants is not none %}
    {{ adapter.apply_lf_grants(relation, lf_grants) }}
  {% endif %}

```
seems we forgot to apply lf_tags and lf_grants for this case

**Review by svdimchenko — APPROVED** [review]

**Review by Jrmyy — APPROVED** [review]

**Pull request merged**